### PR TITLE
Fix breakpoints with updated token structure

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Current `next` branch is broken with mixed usage of `tokens.breakpoints`

### WHAT is this pull request doing?

Updates breakpoint utilities to use either `metaTokens.breakpoints` or `tokens.breakpoints` token structures.

### CODE REVIEW

File: polaris-tokens/src/utilities.ts [hunk 1]
```diff
@@ -118,7 +118,9 @@ export function getCustomPropertyNames(tokens: Tokens) {
     .flat();
 }
 
-export type BreakpointsTokenGroup = typeof breakpointsTokenGroup;
+export type BreakpointsTokenGroup = {
+  [key in keyof typeof breakpointsTokenGroup]: string | TokenGroup;
+};
```
Comment (sam-b-rose) [inline polaris-tokens/src/utilities.ts]: @aaronccasanova thoughts on opening this structure to either token structure (metaTokens or just tokens)?

Or do you think we should have two different types? Maybe something like the following:

```ts
type ExtractValues<T extends { value: string }> = {
  [key in keyof T]: T['value'];
}

export type MetaBreakpointsTokenGroup = typeof breakpointTokenGroup;
export type BreakpointsTokenGroup = ExtractValues<MetaBreakpointTokenGroup>;
```

Comment (aaronccasanova) [reply polaris-tokens/src/utilities.ts]: I think we should keep it scoped to the `breakpoints` token group so we can ensure `polaris-react` pulls in the treeshaken named export.

The last suggestion looks sufficient 👍
```ts
// src/types.ts
type ExtractValues<T extends MetaTokenGroup> = {
  [K in keyof T]: T[K]['value'];
}

// src/utilities.ts
import {Exact, ExtractValues} from './types'
import type {breakpoints as metaBreakpointsTokenGroup} from './token-groups/breakpoints';

export type MetaBreakpointsTokenGroup = typeof metaBreakpointTokenGroup;
export type BreakpointsTokenGroup = ExtractValues<MetaBreakpointTokenGroup>;

export function getMediaConditions<
  T extends Exact<BreakpointsTokenGroup, T>
>(breakpoints: T) {...}
```

Also, I don't mind the dual type exports and think that's a good distinguishing naming convention (which we are already using e.g. `Tokens`/`MetaTokens`, `TokenGroup`/`MetaTokenGroup`, etc.). However, if after the rework we don't need the `MetaBreakpointsTokenGroup` export let's nix it, thoughts?

Comment (sam-b-rose) [reply polaris-tokens/src/utilities.ts]: Alright, updated the types and moved removal of metadata to a utility function 👍 lmk wha you think @aaronccasanova

File: polaris-tokens/src/utilities.ts [hunk 2]
```diff
@@ -118,7 +125,19 @@ export function getCustomPropertyNames(tokens: Tokens) {
...
+export function removeMetadata<T extends Exact<MetaTokenGroup, T>>(
+  tokenGroup: T,
+): ExtractValues<T> {
+  return Object.fromEntries(
+    Object.entries(tokenGroup).map(([tokenName, {value}]) => [
+      tokenName,
+      value,
+    ]),
+  ) as ExtractValues<T>;
+}
```
Comment (aaronccasanova) [inline polaris-tokens/src/utilities.ts]:
```suggestion
export function removeMetadata<T extends Exact<MetaTokenGroup, T>>(
  tokenGroup: T,
) {
  return Object.fromEntries(
    Object.entries(tokenGroup).map((entry): Entry<TokenGroup> => {
      const [tokenName, {value}] = entry as Entry<MetaTokenGroup>;

      return [tokenName, value];
    }),
  ) as ExtractValues<T>;
}
```

Light suggestion to cast the `entry` itself in addition to the final `Object.fromEntries` result. This way we can validate each `.map` return type matches `Entry<TokenGroup>`.

File: polaris-tokens/scripts/toTokenValues.ts [hunk 3]
```diff
@@ -1,32 +1,20 @@
...
-        Object.entries(tokenGroup).map(([tokenName, {value}]) => [
-          tokenName,
-          value,
-        ]),
-      );
-
-      return [tokenGroupName, tokenGroupValues];
-    },
+    ([tokenGroupName, tokenGroup]) =>
+      [tokenGroupName, removeMetadata(tokenGroup)] as Entry<Tokens>,
```
Comment (aaronccasanova) [inline polaris-tokens/scripts/toTokenValues.ts]:
```suggestion
    (entry): Entry<Tokens> => {
      const [tokenGroupName, tokenGroup] = entry as Entry<MetaTokens>;

      return [tokenGroupName, removeMetadata(tokenGroup)];
    },
```

Similar to the comment below, casting the `entry` itself allows us to validate the `.map` return type matches `Entry<Tokens>`.

File: polaris-tokens/src/utilities.ts (hunk 1 above)
Comment (aaronccasanova) [reply polaris-tokens/src/utilities.ts]: Update looks great! Left a couple light suggestions

**Review by aaronccasanova — APPROVED** [review]

**Pull request merged**